src/main/java/com/metamx/cache/CaffeineCache.java

```diff
@@ -118,11 +119,18 @@ public void close(String namespace)
   public io.druid.client.cache.CacheStats getStats()
   {
     final com.github.benmanes.caffeine.cache.stats.CacheStats stats = cache.stats();
+    final Policy.Eviction eviction = cache.policy().eviction().orElse(null);
+    final long size;
+    if (eviction != null && eviction.isWeighted()) {
+      size = eviction.weightedSize().orElse(-1);
+    } else {
+      size = -1;
+    }
```

Comment:
it's a bit weird to convert to optional to null to check if they're present. why not just check isPresent directly?

Alternatively you could go full functional and embrace the option style (although OptionalLong throws a bit of a wrench in here, because it's not Optional<Long>, in which case we could have used `flatMap`)

```
cache
      .policy()
      .eviction()
      .map(eviction -> eviction.isWeighted() ? eviction.weightedSize() : OptionalLong.empty())
      .orElse(OptionalLong.empty()).orElse(-1);
```